### PR TITLE
Issue #15865: list block-tags for JavadocParagraphCheck

### LINKED ISSUE #15865: JavadocParagraph: define in documentation list of validated block-tags
https://checkstyle.org/checks/javadoc/javadocparagraph.html#Description

Warnings are generated about the presence (or lack of presence) of \<p> tag in javadoc text that is inconsistent with google-java-format-1.24.0.

```
$ cat Test.java 
package org.natuna.test;

/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <div style="padding: 1em 1em 1em">
 * <table>
 * <caption></caption>
 * <thead>
 * <tr>
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Direction
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Abbreviation
 * </tr>
 * </thead>
 * <tbody>
 * <tr>
 * <td>East</td> <td>E</td>
 * </tr>
 * </tbody>
 * </table>
 * </div>
 *
 * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
 */
public class Test {
  /** Default constructor. */
  public Test() {
    super();
  }
}

$ java -jar checkstyle-10.20.0-all.jar -c /google_checks.xml Test.java 
Starting audit...
Audit done.

$ java -jar google-java-format-1.24.0-all-deps.jar Test.java > TestUpdated.java
$ cat TestUpdated.java 
package org.natuna.test;

/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <p><div style="padding: 1em 1em 1em">
 *
 * <table>
 * <caption></caption>
 * <thead>
 * <tr>
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Direction
 * <th scope="col" style="padding: .3em 4em 0 0; text-align: left">Abbreviation
 * </tr>
 * </thead>
 * <tbody>
 * <tr>
 * <td>East</td> <td>E</td>
 * </tr>
 * </tbody>
 * </table>
 *
 * </div>
 *
 * <p>This plugin assumes numbered streets have the correct suffix, when applicable.
 */
public class Test {
  /** Default constructor. */
  public Test() {
    super();
  }
}

$ java -jar checkstyle-10.20.0-all.jar -c /google_checks.xml TestUpdated.java 
Starting audit...
[WARN] /var/tmp/TestUpdated.java:6:4: <p> tag should not precede HTML block-tag
      '<div>', <p> tag should be removed. [JavadocParagraph]
[WARN] /var/tmp/TestUpdated.java:27:1: The name of the outer type and the file do not match. [OuterTypeFilename]
Audit done.
```

Expected: no violation `tag should not precede HTML block-tag '<div>', <p> tag should be removed. [JavadocParagraph]`
Violation from OuterTypeFilename is side effect for formatter output to new file.

### PR DESCRIPTION
**Pull request by romani**
closes  #15865

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java [hunk 1]
```diff
@@ -42,8 +42,14 @@
  * <li>Each paragraph but the first has &lt;p&gt; immediately
  * before the first word, with no space after.</li>
  * <li>First paragraph tag should not precede
- * <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>,
- * nested paragraph tags are allowed to do that.</li>
+ * <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>.
+ * Javadoc support only following block-tags:
```
Comment (Zopsss) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java:46]: The doc says:

```
First paragraph tag should not precede HTML block-tag
```

Here `First paragraph tag` can confuse people. By first paragraph tag, we meant the outer most paragraph tag, not the paragraph tag which comes first in the javadoc. User might misunderstand this statement as:

```
The paragraph tag which occurs first in the javadoc should not be preceded by block-tag, it's sibling paragraph tag can do that
```

The check's behavior:

```java
/**
 * Plugin to standardize all directions to their appropriate abbreviation.
 *
 * <p><div>Test</div></p>
 *
 * <p><h1>Test</h1></p>
 *
 * <table>
 * <tbody>
 * <p>
 * <tr>
 * nested paragraph preceding block tag, this is ok.
 * </tr>
 * </p>
 * </tbody>
 * </table>
 */
public class Test {
  /** Some javadoc. */
  public static void main(String[] args) {

  }
}
```

```
$  java -jar .\checkstyle-10.20.1-all.jar -c .\javadocparagraph_config.xml .\Test.java
Starting audit...
[ERROR] C:\checkstyle testing\.\Test.java:4:4: <p> tag should not precede HTML block-tag '<div>', <p> tag should be removed. [JavadocParagraph]
[ERROR] C:\checkstyle testing\.\Test.java:6:4: <p> tag should not precede HTML block-tag '<h1>', <p> tag should be removed. [JavadocParagraph]
Audit done.
Checkstyle ends with 2 errors.
```

Here the user might not expect violation on:

```
<p><h1>Test</h1></p>
```

We should update the doc to make it more clear, something like:

```
The outer most paragraph tags should not preceded HTML block-level tags. Nested paragraph tags are allowed to do that.
```

Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java]: done.

**Comment by romani** [comment]
GitHub, generate website

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java [hunk 2]
```diff
@@ -41,9 +41,15 @@
...
- * nested paragraph tags are allowed to do that.</li>
+ * <li>The outer most paragraph tags should not preceded
+ * <a href="https://www.w3schools.com/html/html_blocks.asp">HTML block-tag</a>.
+ * Javadoc support only following block-tags:
+ * &lt;address&gt;,&lt;blockquote&gt;
+ * ,&lt;div&gt;,&lt;dl&gt;
+ * ,&lt;h1&gt;,&lt;h2&gt;,&lt;h3&gt;,&lt;h4&gt;,&lt;h5&gt;,&lt;h6&gt;,&lt;hr&gt;
+ * ,&lt;ol&gt;,&lt;p&gt;,&lt;pre&gt;
+ * ,&lt;table&gt;,&lt;ul&gt;.
+ * Nested paragraph tags are allowed to do that.</li>
```
Comment (Zopsss) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocParagraphCheck.java:52]: Please use following: 


```
The outer most paragraph tags should not precede HTML block-tag. Nested paragraph tags are allowed to do that. This check only supports following block-tags: <address>,<blockquote> ,<div>,<dl> ,<h1>,<h2>,<h3>,<h4>,<h5>,<h6>,<hr> ,<ol>,<p>,<pre> ,<table>,<ul>.
```

**Review by Zopsss — CHANGES_REQUESTED** [review]
Last:

**Review by Zopsss — APPROVED** [review]
lgtm. Thanks for the changes!

**Pull request merged**